mypy/messages.py

```diff
@@ -2517,11 +2517,11 @@ def iteration_dependent_errors(self, iter_errors: IterationDependentErrors) -> N
 
 def quote_type_string(type_string: str) -> str:
     """Quotes a type representation for use in messages."""
-    no_quote_regex = r"^<(tuple|union): \d+ items>$"
     if (
         type_string in ["Module", "overloaded function", "<deleted>"]
         or type_string.startswith("Module ")
-        or re.match(no_quote_regex, type_string) is not None
+        or type_string.startswith(("<tuple ", "<union "))
+        and type_string.endswith(" item>")
```

Comment:
Didn't test this but the original regex has a colon after tuple/union and items is plural, did you mean this:
```suggestion
        or type_string.startswith(("<tuple: ", "<union: "))
        and type_string.endswith(" items>")
```

Reply:
Yeah, let's stay closer to original logic (because why not).

Reply:
Hm does the fact that no tests or primer changed with the wrong startswith/endswith mean that this code is dead?